components/brave_federated/adapters/flower_helper.cc

```diff
@@ -0,0 +1,188 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/adapters/flower_helper.h"
+
+#include <sstream>
+#include <utility>
+#include <vector>
+
+#include "base/logging.h"
+#include "base/notreached.h"
+#include "brave/components/brave_federated/task/model.h"
+#include "brave/third_party/flower/src/brave/flwr/serde.h"
+#include "brave/third_party/flower/src/proto/flwr/proto/fleet.pb.h"
+#include "brave/third_party/flower/src/proto/flwr/proto/node.pb.h"
+#include "brave/third_party/flower/src/proto/flwr/proto/transport.pb.h"
+
+namespace brave_federated {
+
+std::string BuildGetTasksPayload() {
+  std::string request;
+
+  flower::Node node;
+  node.set_node_id(0);
+  node.set_anonymous(true);
+
+  flower::PullTaskInsRequest pull_task_instructions_request;
+  *pull_task_instructions_request.mutable_node() = node;
+  pull_task_instructions_request.add_task_ids("0");
+  pull_task_instructions_request.SerializeToString(&request);
+
+  return request;
+}
+
+absl::optional<Task> ParseTask(const flower::TaskIns& task_instruction) {
+  const std::string& id = task_instruction.task_id();
+  const std::string& group_id = task_instruction.group_id();
+  const std::string& workload_id = task_instruction.workload_id();
+  if (id.empty() || group_id.empty() || workload_id.empty()) {
+    VLOG(2) << "Invalid task id received from FL service";
+    return absl::nullopt;
+  }
+  const TaskId& task_id = {
+      .id = id, .group_id = group_id, .family_id = workload_id};
```

Comment:
Should be constructed just before use, at the very end of this function

Reply:
used the TaskId struct to include input validation logic to avoid unnecessary copying